client/src/components/Calendar.js

```diff
@@ -0,0 +1,183 @@
+    : totalDaysOfEachMonth;
+
+  return (
+    <Modal>
+      <div className='modal_open'>
+        <CalendarView>
+          <Header>
+            <Button onClick={() => setDate(new Date(year, month - 1, day))}>
+              <MdNavigateBefore className='icon' />
+            </Button>
+            <div>
+              {months[month]} {year}
+            </div>
+            <Button onClick={() => setDate(new Date(year, month + 1, day))}>
+              <MdNavigateNext className='icon' />
+            </Button>
+          </Header>
+          <Body>
+            {weekdays.map((d) => (
+              <Day key={d}>
+                <strong>{d}</strong>
+              </Day>
+            ))}
+            {Array(days[month] + (startDay - 1))
+              .fill(null)
+              .map((_, index) => {
+                const d = index - (startDay - 2);
+                const formatted = `${d}/${month + 1}/${year}`;
+                return (
+                  <Day
+                    key={index}
+                    isToday={d === today.getDate()}
+                    isSelected={d === day}
+                    onClick={() => {
+                      setDate(new Date(year, month, d));
+                      onSetSelectDate(formatted);
+                    }}
+                  >
+                    {d > 0 ? d : ''}
+                  </Day>
+                );
+              })}
+          </Body>
+        </CalendarView>
+        <ButtonClose onClick={() => onSetShowCalendar(false)}>OK</ButtonClose>
+      </div>
+      <div className='modal_open--overlay'></div>
+    </Modal>
+  );
+}
+
+/* Calendar.propTypes = {
+  onSetShowCalendar: PropTypes.func,
+  onSetSelectDate: PropTypes.func,
+}; */
+
+const Modal = styled.div`
+  position: relative;
+  .modal_open {
+    display: flex;
+    flex-flow: column wrap;
+    background-color: var(--grey-lightest);
+    border-radius: 0.5rem;
+    max-width: 26rem;
+    margin: auto;
+    padding: 1rem;
+    position: absolute;
+    z-index: 999;
+  }
+  .modal_open--overlay {
+    background-color: rgba(0, 0, 0, 0.5);
+    position: fixed;
+    top: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+    z-index: 99;
+  }
+`;
+
+const CalendarView = styled.section`
+  display: grid;
+`;
+
+const Header = styled.header`
+  display: flex;
+  justify-content: space-between;
+  background-color: var(--primary-dark);
+  color: var(--grey-lightest);
+  font-weight: bold;
+  padding: 0.5rem 0.5rem 0.25rem 0.5rem;
+`;
+
+const Button = styled.button`
+  background-color: transparent;
+  border: none;
+  color: var(--grey-lightest);
+  .icon {
+    font-size: 2rem;
+  }
+  cursor: pointer;
+`;
+
+const Body = styled.article`
+  width: 100%;
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const Day = styled.div`
+  background-color: ${(props) => props.isSelected && 'var(--secondary-light)'};
+  border: ${(props) => props.isToday && '1px solid var(--grey-lightest)'};
+  display: flex;
```

Comment:
you've always started sorting by putting display first, but here you put it in alphabetical order.  To be consistent, you should put it at the top as well :)